pkg/sql/sem/tree/constant.go

```diff
@@ -168,6 +168,8 @@ func (expr *NumVal) Format(ctx *FmtCtx) {
 	s := expr.origString
 	if s == "" {
 		s = expr.value.String()
+	} else if strings.EqualFold(s, "NaN") {
+		s = "'NaN'"
```

Comment:
you need to check `FmtBareIdentifiers` to print the `'` though, right?

Reply:
in this case it'd be `FmtParsableNumerics`

but i don't think we want it here -- I think a `NaN` NumVal can only appear in the context of formatting a query string. if we normalized `SELECT a IS NAN` into `SELECT a = NaN` it would actually be a syntax error.